### PR TITLE
✨ [video cache] Append captions track from cache response

### PR DESCRIPTION
**Pull request by processprocess**
The video cache will respond with `captions_url` (not implemented at the time of this PR).
If the response contains `captions_url` and the video does not already have a `track` child, a `track` element will be appended to the video with `src` and `type`.

### CODE REVIEW

**Review by mszylkowski — APPROVED** [review]

**Review by gmajoulet — CHANGES_REQUESTED** [review]
This will fail to load on publisher's origins: the captions will load from the `cdn.ampproject.org` URL, while being on the origin URL, resulting in a CORS error.

I believe you will need to set a `crossorigin` attribute on the `<video` tag for it to work. You can test this very easily by setting the `<track src=` pointing to a different domain.

This however needs to happen much earlier in the code process, as the attribute needs to be present BEFORE we send the video cache request, as we use it to determine whether the video URLs need the ACAO header:

https://github.com/ampproject/amphtml/blob/main/extensions/amp-video/0.1/video-cache.js#L233

**Comment by processprocess** [comment]
> I believe you will need to set a crossorigin attribute on the <video tag for it to work.

> This however needs to happen much earlier in the code process, as the attribute needs to be present BEFORE we send the video cache request, as we use it to determine whether the video URLs need the ACAO header

Because of this, wouldn't we need to set `'amp_video_require_acao_header': 1` always?

File: extensions/amp-video/0.1/test/test-video-cache.js [hunk 1]
```diff
@@ -494,6 +502,61 @@ describes.realWin('amp-video cached-sources', {amp: true}, (env) => {
...
+            ],
+          }),
+      });
+      const videoEl = createVideo([{src: 'video.mp4'}]);
+      await fetchCachedSources(videoEl, env.ampdoc);
+
+      const trackEl = videoEl.querySelector('track');
+      expect(trackEl).to.exist;
+    });
+    it('should not append track element if video already has a track child', async () => {
```
Comment (gmajoulet) [inline extensions/amp-video/0.1/test/test-video-cache.js:523]: (thank you for writing such great tests it makes reviewing the UX a lot easier)

maybe we should still render the track element after the existing ones? So if they 404 it falls back to ours

Comment (processprocess) [reply extensions/amp-video/0.1/test/test-video-cache.js:523]: to recap our conversation:
Track elements don't automatically fallback to the next track upon 404. 
For now we will keep this functionality of replacing the track.

**Pull request merged**